### PR TITLE
Add feature-based browser blocking

### LINKED ISSUE #67: Block browsers other than Chrome/Safari/Firefox/Opera etc

### PR DESCRIPTION
**Pull request by Kegsay**
This uses Modernizr to do feature-based detection in browsers. The `.json` file is there as a build file to say "make me a Modernizr which can run these tests" so we don't have redundant tests lying around. The results of the tests are then checked  in `index.js` **before** we ever hook in `MatrixChat`. This neatly side-steps any multiple instantiation problems because only when they click through do we replace the `CompatibilityPage` with the actual app.

This fixes #67

### CODE REVIEW

File: src/skins/vector/views/pages/CompatibilityPage.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    render: function() {
+
+        return (
+        <div className="mx_CompatibilityPage">
+            <div className="mx_CompatibilityPage_box">
+                <p>Sorry, your browser is <b>not</b> able to run Vector.</p>
+                <p>
+                Buttons and images may appear out of place, communication may
+                not be possible and all manner of chaos may be unleashed.
+                </p>
```
Comment (ara4n) [inline src/skins/vector/views/pages/CompatibilityPage.js]: "Please install Chrome from [here] for the best experience"

**Comment by ara4n** [comment]
Related is #291, although that may merit its own PR.

**Comment by ara4n** [comment]
LGTM, other than not solving the WebRTC problem.

**Comment by Kegsay** [comment]
Indeed, doing that as a separate PR :)

**Pull request merged**